source/Projectile.cpp

```diff
@@ -287,35 +295,54 @@ shared_ptr<Ship> Projectile::TargetPtr() const
 
 void Projectile::CheckLock(const Ship &target)
 {
-	double base = hasLock ? 1. : .5;
+	double base = hasLock ? 1. : .15;
 	hasLock = false;
 	
-	// For each tracking type, calculate the probability that a lock will be
-	// lost in a given five-second period. Then, since this check is done every
-	// second, test against the fifth root of that probability.
+	// For each tracking type, calculate the probability twice every second that a
+	// lock will be lost
 	if(weapon->Tracking())
 		hasLock |= Check(weapon->Tracking(), base);
 	
 	// Optical tracking is about 15% for interceptors and 75% for medium warships.
 	if(weapon->OpticalTracking())
 	{
 		double weight = target.Mass() * target.Mass();
-		double probability = weapon->OpticalTracking() * weight / (200000. + weight);
+		double probability = weapon->OpticalTracking() * weight / (150000. + weight);
 		hasLock |= Check(probability, base);
```

Comment:
I have the impression that weapons with more than 1 tracking type (for example radar-track and optical track) actually have a higher chance to lose a lock than weapons with only a single tracking type, because the (single) lock can be lost for each tracking type that is used.

I would normally expect missiles to use the best tracking type for each part of their trajectory and become more resilient against losing target locks if they track over more than 1 type.

This looks like something that already existed in the existing code (before this PR), so I don't consider this a blocker for this PR. But maybe it would be nice for a follow-up PR to perform the hasLock-Check one with the mminimum p



This looks like something that already existed in the existing code, so this PR doesn't make it worse.